transport_plugins/jlink/iotile_transport_jlink/jlink_background.py

```diff
@@ -33,14 +40,22 @@ def __init__(self, jlink_adapter, loop=SharedLoop):
         self._maintenance_task = None
         self.rpc_response_event = self._loop.create_event() # triggered when the controller acknowledge rpc response
 
-    async def send_rpc(self, device_info, control_info, address, rpc_id, payload, timeout):
+    async def send_rpc(self, device_info, control_info, address, rpc_id, payload, timeout, best_effort=False):
         """Write and trigger an RPC."""
 
         write_address, write_data = control_info.format_rpc(address, rpc_id, payload)
         await self.write_memory(write_address, write_data, chunk_size=4)
 
         await self._trigger_rpc(device_info)
-        await self._wait_rpc_completion(control_info, timeout)
+
+        if rpc_id == self.RESET_RPC_ID and address == self.CONTROLLER_ADDRESS:
+            await asyncio.sleep(2)
+            await self._update_state_flags_after_reset(device_info, control_info)
+            return None
+        elif best_effort:
+            return None
+        else:
+            await self._wait_rpc_completion(control_info, timeout)
```

Comment:
I can't comment on the actual line since it's not in this diff but:

- `_wait_rpc_completion` should raise `DeviceAdapterError` not `HardwareError` for timeout and communication issues.

`None` is not a valid return value from `send_rpc`.  Either it should return the `bytes` result of the RPC call or raise an exception because the RPC call did not succeed.

Reply:
I see what you mean, should I replace other `HardwareError`s with `DeviceAdapterError`?

Reply:
Yes but that can be done later.  Let's merge this one and then setup a final round of testing here.